### PR TITLE
Feature/admin add edit user profile

### PR DESCRIPTION
**Pull request by stelli98**
Features that are covered are : 
1.  Add User
2. View User Personal
3. View User Family 
4. Edit User Personal
5. Edit User Family (one by one)
6. Add User Family (Bulk)
7. Delete User Family (one by one)

please refering to this UI : https://www.figma.com/file/RQgoKhbrkH4dSsHJctDvum6n/REIMS?node-id=175%3A2

### CODE REVIEW

File: src/components/js/user-family-detail.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      expandedGroup: []
+    };
+  },
+  computed: {
+    ...mapGetters("user", ["userFamilies"]),
+    userId() {
+      return this.$route.params.id;
+    },
+    disabledAddFamilyButton(){
+      return this.userFamilies.length == 4 ? true : false
```
Comment (wincentrtz) [inline src/components/js/user-family-detail.js]: Ilangin aja ternarynya
Comparator === udh balikkin boolean

File: src/components/js/user-family-detail.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export default {
+  data() {
+    return {
+      expandedGroup: []
+    };
+  },
+  computed: {
+    ...mapGetters("user", ["userFamilies"]),
+    userId() {
+      return this.$route.params.id;
```
Comment (wincentrtz) [inline src/components/js/user-family-detail.js:11]: We can move it to mixin aku rasa ssmua pake hahaha

File: src/mixins/user-form-mixins.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      get() {
+        return this.user.dateOfBirth
+          ? new Date(this.user.dateOfBirth).toISOString()
+          : "";
+      }
+    }
+  },
+  methods: {
+    moveTo(page) {
+      this.$router.push({ name: page });
```
Comment (wincentrtz) [inline src/mixins/user-form-mixins.js]: Ini juga byk dimana2 apakah di mixin aja ya heheh

File: src/mixins/user-form-mixins.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    validateUserForm() {
+      this.$v.user.$touch();
+      if (this.isShowingVehicleField) {
+        this.$v.vehicle.$touch();
+        this.user.vehicle = this.vehicle;
+        if (!this.$v.user.$invalid && !this.$v.vehicle.$invalid) {
+          this.sendCreateUserForm();
+        }
+      } else {
```
Comment (wincentrtz) [inline src/mixins/user-form-mixins.js]: Else if aja ga ush dipisah gini

File: src/router.js [hunk 5]
```diff
@@ -59,27 +59,44 @@ const router = new Router({
     },
     {
       path: "/users",
-      beforeEnter: checkAuthAdmin,
+      // beforeEnter: checkAuthAdmin,
```
Comment (wincentrtz) [inline src/router.js:66]: Delete comment

File: src/views/js/add-family.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    currentDateTime() {
+      return new Date().toISOString();
+    },
+    userId() {
+      return this.$route.params.id;
+    },
+    maxFamilyField() {
+      return 4 - this.userFamilies.length;
+    },
+    isSpouseDataAvailable() {
```
Comment (wincentrtz) [inline src/views/js/add-family.js]: Ini  mw buat return boolean atau ap?

File: src/components/js/user-family-detail.js (hunk 1 above)
Comment (stelli98) [reply src/components/js/user-family-detail.js]: bener juga....

File: src/components/js/user-family-detail.js (hunk 2 above)
Comment (stelli98) [reply src/components/js/user-family-detail.js:11]: tapi isinya userId doang ?

Comment (wincentrtz) [reply src/components/js/user-family-detail.js:11]: lupa submit bbrp review dibawah maybe bs digabungkan,
tapi kalau ndak bs dibuat utk bbrp tipical mixins juga

**Pull request closed without merging**